controller/composite/controller.go

```diff
@@ -241,13 +241,12 @@ func (pc *parentController) enqueueParentObject(obj interface{}) {
 func (pc *parentController) updateParentObject(old, cur interface{}) {
 	// Ignore updates where the ResourceVersion changed (not a resync)
 	// but the spec hasn't changed (e.g. our own status updates).
-	// TODO(metacontroller#26): Use ObservedGeneration instead of this.
 	oldParent := old.(*unstructured.Unstructured)
 	curParent := cur.(*unstructured.Unstructured)
 	if curParent.GetResourceVersion() != oldParent.GetResourceVersion() {
-		oldSpec := k8s.GetNestedField(oldParent.UnstructuredContent(), "spec")
-		curSpec := k8s.GetNestedField(curParent.UnstructuredContent(), "spec")
-		if reflect.DeepEqual(oldSpec, curSpec) {
```

Comment:
As mentioned above, I still recommend leaving this function unchanged for now. Changing this is going to be more complicated that what you have now (we need to fall back to DeepEqual unless we know generation is being incremented) and it's not even worth it because it'll become less reliable (not every built-in resource that has /status increments generation properly).

Reply:
I recalled a suggestion of changing this to `.generation` or `.observedGeneration`. Does it mean that the previous suggestion does not apply anymore?

Reply:
The [original suggestion](https://github.com/GoogleCloudPlatform/metacontroller/pull/102/files#r221055866) was a two-parter: (1) this should be generation instead of observedGeneration, and (2) there's more you'll need in addition to this change, and it will become less reliable, so let's not change anything here right now. Sorry for the confusing structure. I should have started with the final recommendation instead of burying it in the middle of a long comment.

Reply:
Noted, I will revert this function.